Add --version flag to output current version

mhogerheijde opened the pull request:
Fixes #1766

Adds a new flag to enable version output of the current `otp.jar`.

`--version` enables output of version and prints all version bits by default.

```
$ java -jar otp.jar --version
version: 1.0.0-SNAPSHOT
major: 1
minor: 0
patch: 0
qualifier: SNAPSHOT
commit: 5baeb0da96f679cfb7c6d7b8dea192a127f83b35
```

mhogerheijde commented on src/main/java/org/opentripplanner/common/MavenVersion.java:
```diff
@@ -111,6 +113,30 @@ public String toStringVerbose() {
         return String.format("%s => %s UID=%d", version, this.toString(), getUID());
     }
 
+    public String getVersion(String fragment) {
```
I was in doubt about using an `enum` for this. If anyone objects to using a `String` here, I will happily create an `enum`.

mhogerheijde commented on src/main/java/org/opentripplanner/standalone/CommandLineParameters.java:
```diff
@@ -70,6 +69,15 @@
+    @Parameter(names = { "--version-fragment", },
+            description = "Print the a fragment of the version, and then exit. Use one of [`version`, `major`, `minor`, `incremental`, " +
+                    "`qualifier`, `commit` ] to specify which fragment")
+    public String version = DEFAULT_VERSION_FRAGMENT;
```
I wanted to combine the two allowing for `--version` and `--version commit`, but as far as I could find, this is not possible. Setting `variableArity = true` makes `jCommander` still expect a minimum of 1 value. Setting `arity = 0` makes it impossible to pass any values. I would like to say something like `arity = {0, 1}`.

mhogerheijde replied:
P.S. I should have renamed the variables to `version` and `versionFragment` resp.

mhogerheijde commented on src/main/java/org/opentripplanner/standalone/CommandLineParameters.java:
```diff
@@ -70,6 +69,15 @@
+    @Parameter(names = { "--version-fragment" },
+            description = "Print the a fragment of the version, and then exit. Use one of [`version`, `major`, `minor`, `incremental`, " +
+                    "`qualifier`, `commit` ] to specify which fragment")
+    public String versionFragment = DEFAULT_VERSION_FRAGMENT;
```
I wanted to combine the two allowing for `--version` and `--version commit`, but as far as I could find, this is not possible. Setting `variableArity = true` makes `jCommander` still expect a minimum of 1 value. Setting `arity = 0` makes it impossible to pass any values. I would like to say something like `arity = {0, 1}`.

abyrd replied:
Do we really need an option to select the fragment? If we just always output all version fragments, it's very easy to machine-parse and pull out the elements we need since each element is on a separate line and begins with the name of the fragment.

mhogerheijde replied:
_need_ would indeed be a stretch. It makes things a bit easier over doing somethimg like 
```
java -jar otp.jar --version | grep"commit" | cut -c8-
```
Which isn't that hard to do. Especially because I coudn't combine the two I see that it might be overkill.

abyrd commented on src/main/java/org/opentripplanner/standalone/OTPMain.java:
```diff
@@ -64,6 +70,18 @@ public static void main(String[] args) {
+
+            if (params.version) {
+                // Disable logging, we don't want to see anything else than the intended output below.
+                ch.qos.logback.classic.Logger mavenVersionLogger =
```
Programmatically shutting off logging (which is kind of ugly) should not be necessary any more, I silenced the logging that was happening every time MavenVersion parsed the properties.

mhogerheijde replied:
I'll take it out. Coudn't figure out any other way of disabling log output for this without affecting the default.

mhogerheijde commented:
@abyrd Sorry that it took quite a while to update this.

I've removed the support for --version-fragment, and removed the code that disabled logging.
